sample_project/Source/sample_project/Feature Layer/FeatureLayer.cpp

```diff
@@ -38,51 +38,93 @@ void AFeatureLayer::OnResponseRecieved(FHttpRequestPtr Request, FHttpResponsePtr
-				auto type = feature->GetObjectField("geometry")->GetStringField("type");
-				//this will get the geometry or coordinates of the feature
-				auto coordinates = feature->GetObjectField("geometry")->GetArrayField("coordinates");
+					//create a new feature object to store the data received associated with this feature iteration
+					FFeatureLayerProperties featureLayerProperties;
+					auto feature = Features[i]->AsObject();
+					//get the properties field associated with the individual feature
 
-				//To avoid crashes, this checks to see if the type of feature is Point, if so it will get the geometry
-				//if not, it will return an error
-				//current the only type of data supported by this sample is Point Layers, but more will be added in the future.
-				if(type.ToLower() == "point")
-				{
-					for (const auto Coordinate : coordinates)
+					/*outfield can be set in the scene on bp_feature
+					this loop will take each outfield set in the scene and check to see if the outfield exists
+					if it does exist, it will return the result of the outfield associated with this feature
+					if it does not exist, it will return and error message in the scene*/
+					if(bGetAllOutfields)
+					{
+						for (auto outfield : WebLink.OutFields)
+						{
+							auto propertyOutfield = feature->GetObjectField("properties")->GetStringField(outfield);
+							if(propertyOutfield.IsEmpty())
+							{
+								featureLayerProperties.FeatureProperties.Add(FString::FromInt(feature->GetObjectField("properties")->GetIntegerField(outfield)));
+							}
+							else
+							{
+								featureLayerProperties.FeatureProperties.Add(feature->GetObjectField("properties")->GetStringField(outfield));	
+							}
+						}	
+					}
+					else
 					{
-						featureLayerProperties.GeoProperties.Add(Coordinate->AsNumber());
+						for (auto outfield : OutFieldsToGet)
+						{
+							auto propertyOutfield = feature->GetObjectField("properties")->GetStringField(outfield);
+							if(propertyOutfield.IsEmpty())
+							{
+								featureLayerProperties.FeatureProperties.Add(FString::FromInt(feature->GetObjectField("properties")->GetIntegerField(outfield)));
+							}
+							else
+							{
+								featureLayerProperties.FeatureProperties.Add(feature->GetObjectField("properties")->GetStringField(outfield));	
+							}
+						}	
 					}
-					bCoordinatesErrorReturn = false;
+					//this will get the type of feature
+					auto type = feature->GetObjectField("geometry")->GetStringField("type");
+					//this will get the geometry or coordinates of the feature
+					auto coordinates = feature->GetObjectField("geometry")->GetArrayField("coordinates");
+
+					//To avoid crashes, this checks to see if the type of feature is Point, if so it will get the geometry
+					//if not, it will return an error
+					//current the only type of data supported by this sample is Point Layers, but more will be added in the future.
+					if(type.ToLower() == "point")
```

Comment:
Space after `if`, same for any other occurence
```suggestion
					if (type.ToLower() == "point")
```